### PR TITLE
Fix destroyed fork backpressure.

### PR DESCRIPTION
**Pull request by vqvu**
See https://github.com/caolan/highland/issues/179#issuecomment-160243577 for description.

### CODE REVIEW

File: lib/intMap.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+var hasOwn = Object.prototype.hasOwnProperty;
+
+/**
+ * A very simple polyfill for Map with limited functionality
+ * that only works for integer keys.
+ */
+function IntMap() {
+    this.map = {};
```
Comment (svozza) [inline lib/intMap.js]: Not sure if there's much point but you could reuse the isES5 in `index.js` function to see whether you could use `Object.create(null)` here.

**Comment by svozza** [comment]
Very nice implementation.

File: lib/intMap.js (hunk 1 above)
Comment (vqvu) [inline lib/intMap.js]: I didn't do it because I'm using `Object.prototype.hasOwnProperty`. Is there a perf improvement to doing `Object.create(null)`?

Comment (svozza) [inline lib/intMap.js]: The perf gain is in not having to do the `hasOwnProperty` check but tbh like most micro-optimisations I doubt it's worth the hassle.

Comment (vqvu) [inline lib/intMap.js]: Ah, gotcha. Agreed that it's not worth the hassle. Modern browsers and Nodes will just end up using `global.Map` anyway.

**Pull request merged**